File: midlab3/MP01.java
package midlab3;

import java.io.*;

public class MP01 {
    public static void main(String[] args) {

        String filePath = "Sample_Data-Prog-2-csv.csv";
        int recordCount = 0;

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {

            String line;

            while ((line = br.readLine()) != null) {
                line = line.trim();

                if (line.startsWith("\"")) {   // valid candidate row
                    recordCount++;
                }
            }

            System.out.println("Total number of records: " + recordCount);

        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }
}

File: midlab3/MP01.js
const fs = require("fs");

const filePath = "Sample_Data-Prog-2-csv.csv";
let recordCount = 0;

try {
    const data = fs.readFileSync(filePath, "utf8");
    const lines = data.split("\n");

    for (let line of lines) {
        line = line.trim();

        if (line.startsWith('"')) { // valid candidate row
            recordCount++;
        }
    }

    console.log("Total number of records:", recordCount);

} catch (err) {
    console.log("Error reading file:", err.message);
}